src/main/java/com/github/mauricioaniche/ck/CKVisitor.java

```diff
@@ -131,7 +134,7 @@ public boolean visit(MethodDeclaration node) {
 		// so, let's call them
 		classes.peek().classLevelMetrics.stream().map(metric -> (CKASTVisitor) metric).forEach(ast -> ast.visit(node));
 		if(!classes.peek().methods.isEmpty())
-			classes.peek().methods.peek().methodLevelMetrics.stream().map(metric -> (CKASTVisitor) metric).forEach(ast -> ast.visit(node));
+			classes.peek().methods.peek().methodLevelMetrics.stream().map(metric -> (CKASTVisitor) metric).forEach(ast -> {((MethodLevelMetric)ast).setMethodName(currentQualifiedMethodName); ast.visit(node);});
```

Comment:
Same question for the method: why setting its name here?

Reply:
I set the name of the method here because of the same reason I have presented for class.

Reply:
Resolving this one, so we keep the discussion in one comment only.